util/pkg/vfs/vfssync.go

```diff
@@ -213,3 +213,69 @@ func hashesMatch(src, dest Path) (bool, error) {
 	glog.Infof("No compatible hash: %s and %s", src, dest)
 	return false, nil
 }
+
+// CopyTree copies all files in src to dest.  It copies the whole recursive subtree of files.
+func CopyTree(src Path, dest Path) error {
+	srcFiles, err := src.ReadTree()
+	if err != nil {
+		return fmt.Errorf("error reading source directory %q: %v", src, err)
+	}
+
+	destFiles, err := dest.ReadTree()
+	if err != nil {
+		return fmt.Errorf("error reading source directory %q: %v", src, err)
+	}
+
+	destFileMap := make(map[string]Path)
+	for _, destFile := range destFiles {
+		relativePath, err := RelativePath(dest, destFile)
+		if err != nil {
+			return err
+		}
+
+		destFileMap[relativePath] = destFile
+	}
+
+	for _, srcFile := range srcFiles {
+		relativePath, err := RelativePath(src, srcFile)
+		if err != nil {
+			return err
```

Comment:
I cannot find RelativePath in your source tree.  github search is not being kind.  You call if it would be helpful to fmt.Errorf more in here.